system-env-change-listener/src/main/java/biz/netcentric/aem/sysenv/impl/listener/PackageReinstallerMBeanImpl.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * (C) Copyright 2018 Netcentric, A Cognizant Digital Business.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package biz.netcentric.aem.sysenv.impl.listener;
+
+import javax.management.DynamicMBean;
+import javax.management.NotCompliantMBeanException;
+
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.jackrabbit.oak.commons.jmx.AnnotatedStandardMBean;
+
+import biz.netcentric.aem.sysenv.PackageReinstallerMBean;
+
+@Service({DynamicMBean.class})
+@org.apache.felix.scr.annotations.Property(name = "jmx.objectname", value = "biz.netcentric.aem.sysenv:type=Env Specific Package Installer")
+@Component
+public class PackageReinstallerMBeanImpl extends AnnotatedStandardMBean implements PackageReinstallerMBean {
```

Comment:
If you really want to rely on annotations rather use https://helpx.adobe.com/experience-manager/6-5/sites/developing/using/reference-materials/javadoc/com/adobe/granite/jmx/annotation/package-summary.html which should be much more stable. Also please use the lowest version still compatible and add some prerequisite section in the readme to outline with which AEM versions this is compatible.

Reply:
I tried to avoid adding a dependency to Adobe's code since it hasnt done before, isnt that a problem?  I think I would have to depend on the uber jar

Reply:
Just clarify which is the minimum AEM version supported. There are already other dependencies which makes it unlikely to use the hook outside of AEM: https://github.com/Netcentric/system-env-install-hook/blob/develop/apply-system-env-install-hook/pom.xml. IMHO there is no real requirement to support plain Sling.